### PR TITLE
Fixing issue where module mapping didn't match the correct relative directory

### LINKED ISSUE #12: Broken nested module functionality
Hey,

Latest version of the package introduce nested module functionality.

But the same won't work in following cases:
```
"~project/root": ".",
"~project/app": "./app/",
"~project/dir": "./deep/nested/directory/",
```
Although this works just fine:
`"~project/config": "./config/index.js",`

It's pretty clear that module is not working correctly for nested directories. Can we fix this?

Here's the problem
https://github.com/Rush/link-module-alias/blob/d3a5fc9012bad017d45f008c95e000b920163d46/index.js#L178-L179

### PR DESCRIPTION
**Pull request by kwburnett**
@Rush I'm not sure if this fixes #12 or not, but I did notice in my testing that the module names weren't being mapped correctly. Making this change seemed to resolve the issue in my local testing.

However, this should be working for everyone using this plug-in, so I'm not sure if this is the right fix or not. I made this match what was happening here:
https://github.com/Rush/link-module-alias/blob/d3a5fc9012bad017d45f008c95e000b920163d46/index.js#L163

Anyways, let me know your thoughts or if this is a breaking change and something else should be considered.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ async function linkModule(moduleName) {
         }
       }
     }
-    await symlink(path.join('../', target), moduleDir, DIR_LINK_TYPE);
```
Comment (Rush) [inline index.js:177]: Would need to understand the change better in order to approve ...

**Comment by kwburnett** [comment]
Sure thing. I've created a repo I was using to test this, which is available [here](https://github.com/kwburnett/node-tests/tree/master).

When I installed the code, I noticed I didn't have a problem with mapping whether the target ended with a `/` or not. However, when I went to check the symlinks in `node_modules/~project`, I noticed the following:
```
 Directory of C:\source\node-tests\node_modules\~project

04/16/2019  08:01 PM    <DIR>          .
04/16/2019  08:01 PM    <DIR>          ..
04/16/2019  08:01 PM    <JUNCTION>     app [C:\source\node-tests\node_modules\src\app]
04/16/2019  08:01 PM    <DIR>          config
04/16/2019  08:01 PM    <JUNCTION>     dir [C:\source\node-tests\node_modules\src\deep\nested\directory]
```

The symlinks hadn't matched the appropriate directories, so I added an extra `../` to fix this. If I install the repo I linked and change the `index.js` file under `node_modules/link-module-alias` to add the additional `../` matching this commit, I see this when I check the symlinks in the same directory, which is what I expected:
```
 Directory of C:\source\node-tests\node_modules\~project

04/16/2019  08:06 PM    <DIR>          .
04/16/2019  08:06 PM    <DIR>          ..
04/16/2019  08:06 PM    <JUNCTION>     app [C:\source\node-tests\src\app]
04/16/2019  08:06 PM    <DIR>          config
04/16/2019  08:06 PM    <JUNCTION>     dir [C:\source\node-tests\src\deep\nested\directory]
```

I actually couldn't reproduce #13 , but I think this would fix whatever the underlying problem is.

File: index.js (hunk 1 above)
Comment (kwburnett) [reply index.js:177]: See below.

**Comment by Rush** [comment]
Is this PR still relevant?

**Comment by kwburnett** [comment]
Hey man, long time! I just reviewed the code and I think it's still relevant. The issue is caused by nested directories in the module alias. However, the previous code didn't handle more than one directory deep, and actually broke regular functionality. I made some changes to handle these issues.

For example, if alias names don't have a `/` in them, the normal `../` on line 180 works. However, if the alias becomes `this/test` or `this/deep/test`, the symlink using `../` is no longer enough. The new update I made adds a `../` for each nested alias directory. Thoughts?

**Comment by kwburnett** [comment]
Also, if you'd like me to add some unit tests (I'm open to any library you prefer - I've been using Jest quite a bit recently) to ensure things work, I'd be happy to write them. :-)

**Pull request merged**